src/Aspire.Hosting.Azure/AzureWebPubSubResource.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Azure;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an Azure Web PubSub resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class AzureWebPubSubResource(string name) :
+    AzureBicepResource(name, templateResourceName: "Aspire.Hosting.Azure.Bicep.webpubsub.bicep"),
+    IResourceWithConnectionString
+{
+    /// <summary>
+    /// Gets the "endpoint" output reference from the bicep template for Azure Web PubSub.
+    /// </summary>
+    public BicepOutputReference Endpoint => new("endpoint", this);
+
+    /// <summary>
```

Comment:
@vicancy Can you also add `GetConnectionStringAsync`? and use `Endpoint.GetValueAsync` This will make the new local provisioning experience work well.